src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py

```diff
@@ -0,0 +1,233 @@
+    Path(corpus_dir).mkdir(exist_ok=True)
+    arguments.append(f'--corpus_dir={corpus_dir}')
+
+    # Download sanitized binaries.
+    sanitized_binaries = glob.glob(
+        os.path.join(build_dir, '*_*', os.path.basename(target_path)))
+    arguments.append(f'--extra_binaries={",".join(sanitized_binaries)}')
+
+    return engine.FuzzOptions(corpus_dir, arguments, {})
+
+  def fuzz(self, target_path, options, reproducers_dir, max_time):  # pylint: disable=unused-argument
+    """Run a fuzz session.
+
+    Args:
+      target_path: Path to the target.
+      options: The FuzzOptions object returned by prepare().
+      reproducers_dir: The directory to put reproducers in when crashes
+          are found.
+      max_time: Maximum allowed time for the fuzzing to run.
+
+   Returns:
+      A FuzzResult object.
+    """
+    runner = _get_runner()
+    arguments = _DEFAULT_ARGUMENTS[:]
+    arguments.extend(options.arguments)
+    arguments.extend([
+        f'--binary={target_path}',
+    ])
+
+    fuzz_result = runner.run_and_wait(
+        additional_args=arguments, timeout=max_time + _CLEAN_EXIT_SECS)
+    log_lines = fuzz_result.output.splitlines()
+    fuzz_logs = '\n'.join(log_lines)
+
+    crashes = []
+    # Stats report is not available in Centipede yet
+    stats = None
+    for line in log_lines:
+      reproducer_path = _get_reproducer_path(line)
+      if reproducer_path:
+
+        crashes.append(
+            engine.Crash(reproducer_path, fuzz_logs, [],
+                         int(fuzz_result.time_executed)))
+        continue
+
+      #stats = _get_stats(line)
+
+    #if stats is None:
+    #  stats = {}
+
+    return engine.FuzzResult(fuzz_result.output, fuzz_result.command, crashes,
+                             stats, fuzz_result.time_executed)
+
+  def reproduce(self, target_path, input_path, arguments, max_time):  # pylint: disable=unused-argument
+    """Reproduce a crash given an input.
+
+    Args:
+      target_path: Path to the target.
+      input_path: Path to the reproducer input.
+      arguments: Additional arguments needed for reproduction.
+      max_time: Maximum allowed time for the reproduction.
+
+    Returns:
+      A ReproduceResult.
+    """
+    os.chmod(target_path, 0o775)
+    runner = new_process.UnicodeProcessRunner(target_path)
+    with open(input_path, 'rb') as f:
+      result = runner.run_and_wait(timeout=max_time, stdin=f)
+
+    return engine.ReproduceResult(result.command, result.return_code,
+                                  result.time_executed, result.output)
+
+  def _create_temp_dir(self, name):
+    """Creates temporary corpus directory."""
+    new_directory = os.path.join(fuzzer_utils.get_temp_dir(), name)
+    engine_common.recreate_directory(new_directory)
+    return new_directory
+
+  def minimize_corpus(self, target_path, arguments, input_dirs, output_dir,
+                      reproducers_dir, max_time):
+    """Optional (but recommended): run corpus minimization.
```

Comment:
Maybe remove "(but recommended)" I think it makes sense in the parent class but not here.